primitives/core/src/offchain/mod.rs

```diff
@@ -179,6 +179,119 @@ impl TryFrom<u32> for HttpRequestStatus {
 	}
 }
 
+/// Opaque type for offchain timers.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, RuntimeDebug, Encode, Decode, PassByInner)]
+#[cfg_attr(feature = "std", derive(Hash))]
+pub struct TimerId(pub u16);
+
+impl From<TimerId> for u32 {
+	fn from(c: TimerId) -> Self {
+		c.0 as u32
+	}
+}
+
+/// Identifies a kind for pollable value, such as HTTP request or scheduled timers.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[repr(u32)]
+pub enum PollableKind {
+	/// Identifies a HTTP request.
+	Http = 1,
+	/// Identifies a scheduled timer.
+	Timer = 2,
+	/// Unknown pollable kind.
+	Unknown,
+}
+
+impl From<u32> for PollableKind {
+	fn from(value: u32) -> Self {
+		match value {
+			1 => PollableKind::Http,
+			2 => PollableKind::Timer,
+			_ => PollableKind::Unknown,
+		}
+	}
+}
+
+/// Opaque type for pollable values.
+///
+/// This is the core type that allows using async/await in the offchain worker (OCW) context.
+///
+/// At the moment of writing, the OCWs have no way to perform background
+/// computation or scheduling on their own. To work around that, we schedule the
+/// tasks on the host side via runtime-called functions. Doing so returns a handle
+/// ([`PollableId`]) that can be used across host/runtime boundary to poll for results.
+///
+/// The IDs can be manually awaited using the low-level [`sp_io::offchain::pollable_wait`] API.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Encode, Decode, PassByInner)]
+#[cfg_attr(feature = "std", derive(Hash))]
+#[repr(transparent)]
+pub struct PollableId(pub u64);
+
+impl core::fmt::Debug for PollableId {
```

Comment:
This should probably be conditional, we try to avoid too many strings in the runtime.